From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: BillyGalbreath <Blake.Galbreath@GMail.com>
Date: Sun, 23 Sep 2018 20:59:53 -0500
Subject: [PATCH] (Paper) Honor EntityAgeable.ageLock

Original code by PaperMC, licensed under GPLv3
You can find the original code on https://github.com/PaperMC/Paper
This patch specifically is licensed under MIT (according to Paper's LICENSE.md)

diff --git a/src/main/java/net/minecraft/world/entity/AgeableMob.java b/src/main/java/net/minecraft/world/entity/AgeableMob.java
index c0780a8714808498390282b7fa1da1f3aacf8e2a..d165117d62fe8a55d624966e8c4b626c0f52db39 100644
--- a/src/main/java/net/minecraft/world/entity/AgeableMob.java
+++ b/src/main/java/net/minecraft/world/entity/AgeableMob.java
@@ -84,7 +84,7 @@ public abstract class AgeableMob extends PathfinderMob {
     }
 
     public void ageUp(int age, boolean overGrow) {
-        if (ageLocked) return; // Paper - GH-1459
+        if (this.ageLocked) return; // Paper - GH-1459
         int j = this.getAge();
         int k = j;
 
diff --git a/src/main/java/net/minecraft/world/level/block/entity/BeehiveBlockEntity.java b/src/main/java/net/minecraft/world/level/block/entity/BeehiveBlockEntity.java
index 8484e80a70129fb0358d56efab6fd54798b54e6e..b700f67cd8e143170cf258b941c8091f7fc1a4b1 100644
--- a/src/main/java/net/minecraft/world/level/block/entity/BeehiveBlockEntity.java
+++ b/src/main/java/net/minecraft/world/level/block/entity/BeehiveBlockEntity.java
@@ -295,6 +295,7 @@ public class BeehiveBlockEntity extends BlockEntity {
     }
 
     private static void setBeeReleaseData(int ticks, Bee bee) {
+        if (!bee.ageLocked) { // Paper - respect age lock
         int j = bee.getAge();
 
         if (j < 0) {
@@ -302,6 +303,7 @@ public class BeehiveBlockEntity extends BlockEntity {
         } else if (j > 0) {
             bee.setAge(Math.max(0, j - ticks));
         }
+        } // Paper - respect age lock
 
         bee.setInLoveTime(Math.max(0, bee.getInLoveTime() - ticks));
     }
